lib/util/update-metadata-builder.js

```diff
@@ -0,0 +1,149 @@
+/**
+ * @fileoverview Utility for building the metadata patch request
+ */
+
+'use strict';
+
+// ------------------------------------------------------------------------------
+// Requirements
+// ------------------------------------------------------------------------------
+
+var assert = require('assert');
+
+/**
+ * Utility for building the metadata patch request for files and folders
+ *
+ * @constructor
+ * @returns {void}
+ */
+function UpdateMetadataBuilder() {
```

Comment:
The word order here is a little odd, since "UpdateMetadata" isn't a noun that can be built.  Consider changing to "MetadataUpdateBuilder"